styles/tool-bar.less

```diff
@@ -1,50 +1,48 @@
+// NOTE: color of buttons on hover needs adjusting
+
 .tool-bar {
-    &.tool-bar-horizontal {
-        position: relative;
-        height: 3rem;
-        z-index: 100;
-        border: none;
+    background: darken(@app-background-color, 0%);
+    border: none;
+    
+    &.tool-bar-top {
         background: @accent-color;
-
-        &.tool-bar-24px {
-            .tool-bar-btn {
-                width: 3rem;
-                height: 3rem;
-                line-height: 3rem;
-                padding: 0;
-                margin: 0;
-                .text-color(@accent-color);
-            }
-        }
-        .tool-bar-spacer {
-            margin: 0 0.5rem;
-            border: none;
-        }
+        .text-color(@accent-color);
     }
+    
     &.tool-bar-vertical {
-        background: darken(@app-background-color, 3.5%);
-        padding: 0;
         width: 3rem;
-        border: none;
-        box-shadow: inset -0.125rem 0 0.125rem rgba(0,0,0,0.05);
-
-        &.tool-bar-24px {
-            .tool-bar-btn {
-                padding: 0;
-                height: 3rem;
-                width: 3rem;
-                margin: 0;
-                border-radius: 0;
-
-                &:first-child {
-                    background: darken(@accent-color, 5%);
-                    .text-color(@accent-color);
-                }
-            }
+        padding-top: 3rem;
+        
+        .tool-bar-btn:first-child {
+            position: absolute;
+            top: 0;
+            z-index: 100;
+            background: @accent-color;
+            .text-color(@accent-color);
         }
-        .tool-bar-spacer {
-            border: none;
-            border-bottom: 1px solid darken(@app-background-color, 5%);
+    }
+    
+    &.tool-bar-horizontal {
+        height: 3rem;
+    }
+    
+    &.tool-bar-16px, &.tool-bar-24px, &.tool-bar-32px {
+        button.tool-bar-btn {
+            width: 3rem;
+            height: 3rem;
+            margin: 0;
+            padding: 0;
+            border-radius: 0;
+            
+            &:before {
+                // NOTE: should be configured by ui scaling setting
+                font-size: 20px;
```

Comment:
Do try to avoid using pixel units and use `rem` instead. This way, we only would have to change the font-size for `:root` and everything will scale with it.

I'm guilty of using pixels some times (mostly on borders) so feel free to remind me to use `rem`s instead :grin: 

BTW: `1rem` is `16px`. Material Design uses an `8dp` grid which translates to `0.5rem`.